### ISSUE TITLE
Provide generation scores for Flan-T5

### ISSUE THREAD

**Issue by Matthieu-Tinycoaching** [OpenNMT/CTranslate2#1094]
Hello,

Actually using Flan-T5 with the `ctranslate2.Translator.translate_iterable` method, I would like to obtain the transition scores of the generated batch outputs.

These transition scores allow to obtain a score for each of the generated output from the batch, and so to compare the quality of generated outputs: https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationMixin.compute_transition_scores

```
tokenizer = T5Tokenizer.from_pretrained("google/flan-t5-large")
model = T5ForConditionalGeneration.from_pretrained("google/flan-t5-large", device_map="auto")

outputs = model.generate(input_ids, max_new_tokens=decoder_max_len, return_dict_in_generate=True, output_scores=True)

transition_scores = model.compute_transition_scores(outputs.sequences, outputs.scores, normalize_logits=False).to("cpu")
input_length = 1 if model.config.is_encoder_decoder else input_ids.shape[1]
# If you sum the generated tokens' scores and apply the length penalty, you'll get the sequence scores.
# Tip: recomputing the scores is only guaranteed to match with `normalize_logits=False`. Depending on the
# use case, you might want to recompute it with `normalize_logits=True`.
output_length = input_length + np.sum(transition_scores.numpy() < 0, axis=1)
length_penalty = model.generation_config.length_penalty
reconstructed_scores = transition_scores.sum(axis=1) / (output_length**length_penalty)

```
Would this be possible within the ctranslate2 library?

**Comment by Matthieu-Tinycoaching** [comment]
Hi @guillaumekln thanks for your feedback.

I suppose the scores you returned are normalized?

When comparing the scores returned by ctranslate2 and those by the code snippet from hugging face I observed some differences. When using the following 2 contexts (first one no link with the question, second linked):

```
context1 = "Le Père Noël (né Thomas Patrick O'Connor ; mai 1947)[1] est un homme politique américain, un moine et un militant de la protection de l'enfance[2]. Il est membre du conseil municipal de North Pole, en Alaska, depuis 2015[3]. Claus a été réélu au conseil en 2019[4], avec 100 voix[5]. Il est actuellement le maire pro tem[6]. Il a été candidat à la Chambre des représentants des États-Unis lors de l'élection spéciale de 2022. Claus est un moine chrétien de l'ordre anglican celtique Anam Cara[8]. Il a changé son nom pour correspondre au personnage légendaire en 2005, afin d'aider à faire connaître son activisme pour la santé et le bien-être des enfants[9]. Claus est un partisan du socialisme démocratique, y compris le soutien à l'assurance-maladie pour tous, l'aide aux coronavirus, un impôt sur la fortune et l'annulation des prêts étudiants[10]."

context2 = "Des observations de 2015 par la sonde Dawn ont confirmé qu'elle possède une forme sphérique, à la différence des corps plus petits qui ont une forme irrégulière. Sa surface est probablement composée d'un mélange de glace d'eau et de divers minéraux hydratés (notamment des carbonates et de l'argile), et de la matière organique a été décelée. Il semble que Cérès possède un noyau rocheux et un manteau de glace. Elle pourrait héberger un océan d'eau liquide, ce qui en fait une piste pour la recherche de vie extraterrestre. Cérès est entourée d'une atmosphère ténue contenant de la vapeur d'eau, dont deux geysers, ce qui a été confirmé le 22 janvier 2014 par l'observatoire spatial Herschel de l'Agence spatiale européenne."

question = "Quelle caractéristique possède Cérès qui rendrait la vie extraterrestre possible ?"

input_text = [f"Given the context please answer the question. Use only French words. Context: {context1}; Question: {question}; Answer:",
              f"Given the context please answer the question. Use only French words. Context: {context2}; Question: {question}; Answer:"]

translator = ctranslate2.Translator(model_path, device="cuda")
tokenizer = transformers.AutoTokenizer.from_pretrained(model_id)

sentences_tokenized = [tokenizer.convert_ids_to_tokens(tokenizer.encode(sentence)) for sentence in input_text]

results = translator.translate_iterable(sentences_tokenized, max_batch_size=32, beam_size=beam_size, return_scores=True)
print(list(results))
```
I got the following results:
`[TranslationResult(hypotheses=[['▁Le', '▁P', 'ère', '▁No', '<unk>', 'l']], scores=[-0.5016885995864868], attention=[]), TranslationResult(hypotheses=[['▁un', '▁no', 'y', 'au', '▁', 'roch', 'eux', '▁', 'e', 't', '▁un', '▁man', 't', 'eau', '▁de', '▁glace', '.', '▁Elle', '▁pourrait', '▁', 'h', 'é', 'berger', '▁un', '▁', 'océan', '▁', 'd', "'", 'eau', '▁liquid', 'e', ',', '▁ce', '▁qui', '▁', 'en', '▁fait', '▁une', '▁piste', '▁pour', '▁la', '▁recherche', '▁de', '▁vie', '▁extra', 'ter', 'rest', 're', '.']], scores=[-0.05741289258003235], attention=[])]`

But when using the code snippet from HF I got the following results and scores:
```
['<pad> Le Père No<unk> l</s><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad><pad>', "<pad> un noyau rocheux et un manteau de glace. Elle pourrait héberger un océan d'eau liquide, ce qui en fait une piste pour la recherche de vie extraterrestre.</s>"]
tensor([-37.1693,  -0.0589], dtype=torch.float64)
```

The result from the first context with no link has a score far more negative than this provided by ctranslate2.

Do you have an advice?

**Comment by guillaumekln** [comment]
> I suppose the scores you returned are normalized?

Yes.

> The result from the first context with no link has a score far more negative than this provided by ctranslate2.

Most probably you did not consider the different output length when computing the HF scores. The tokens after `</s>` should be ignored and not included in the final score.

**Comment by Matthieu-Tinycoaching** [comment]
Thanks!
